patches/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -0,0 +1,650 @@
+ * and invoke one of the static creation methods, followed by an instance mutator if needed.
+ *
+ * <pre>{@code import static org.jclouds.docker.compute.options.DockerTemplateOptions.Builder.*;
+ *
+ * ComputeService api = // get connection
+ * templateBuilder.options(inboundPorts(22, 80, 8080, 443));
+ * Set<? extends NodeMetadata> set = api.createNodesInGroup(tag, 2, templateBuilder.build());}</pre>
+ */
+public class DockerTemplateOptions extends TemplateOptions implements Cloneable {
+
+   protected List<String> dns = ImmutableList.of();
+   protected String hostname;
+   protected Integer memory;
+   protected Integer cpuShares;
+   protected List<String> entrypoint = ImmutableList.of();
+   protected List<String> commands = ImmutableList.of();
+   protected Map<String, String> volumes = ImmutableMap.of();
+   protected List<String> env = ImmutableList.of();
+   protected Map<Integer, Integer> portBindings = ImmutableMap.of();
+   protected String networkMode;
+   protected Map<String, String> extraHosts = ImmutableMap.of();
+   protected List<String> volumesFrom = ImmutableList.of();
+
+   @Override
+   public DockerTemplateOptions clone() {
+      DockerTemplateOptions options = new DockerTemplateOptions();
+      copyTo(options);
+      return options;
+   }
+
+   @Override
+   public void copyTo(TemplateOptions to) {
+      super.copyTo(to);
+      if (to instanceof DockerTemplateOptions) {
+         DockerTemplateOptions eTo = DockerTemplateOptions.class.cast(to);
+         eTo.volumes(volumes);
+         eTo.hostname(hostname);
+         eTo.dns(dns);
+         eTo.memory(memory);
+         eTo.cpuShares(cpuShares);
+         eTo.entrypoint(entrypoint);
+         eTo.commands(commands);
+         eTo.env(env);
+         eTo.portBindings(portBindings);
+         eTo.networkMode(networkMode);
+         eTo.extraHosts(extraHosts);
+         eTo.volumesFrom(volumesFrom);
+      }
+   }
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o)
+         return true;
+      if (o == null || getClass() != o.getClass())
+         return false;
+      DockerTemplateOptions that = DockerTemplateOptions.class.cast(o);
+      return super.equals(that) && equal(this.volumes, that.volumes) &&
+              equal(this.hostname, that.hostname) &&
+              equal(this.dns, that.dns) &&
+              equal(this.memory, that.memory) &&
+              equal(this.entrypoint, that.entrypoint) &&
+              equal(this.commands, that.commands) &&
+              equal(this.cpuShares, that.cpuShares) &&
+              equal(this.env, that.env) &&
+              equal(this.portBindings, that.portBindings) &&
+              equal(this.extraHosts, that.extraHosts) &&
+              equal(this.volumesFrom, that.volumesFrom);
+   }
+
+   @Override
+   public int hashCode() {
+      return Objects.hashCode(super.hashCode(), volumes, hostname, dns, memory, entrypoint, commands, cpuShares, env, portBindings, extraHosts);
```

Comment:
Oh, right. It's not in `hashCode` - can't fix that here, will maybe get it in with the AutoValue builder PR